dexcom/data_range_test.go

```diff
@@ -0,0 +1,121 @@
+package dexcom_test
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	"github.com/tidepool-org/platform/dexcom"
+	"github.com/tidepool-org/platform/dexcom/test"
+	"github.com/tidepool-org/platform/structure/validator"
+)
+
+var _ = Describe("DataRange", func() {
+	Describe("Validate", func() {
+		DescribeTable("errors when",
+			func(setupDataRangeFunc func() *dexcom.DataRange) {
+				testDataRange := setupDataRangeFunc()
+				validator := validator.New()
+				testDataRange.Validate(validator)
+				Expect(validator.Error()).To(HaveOccurred())
+			},
+			Entry("required end is not set", func() *dexcom.DataRange {
+				dataRange := test.RandomDataRange()
+				dataRange.End = nil
+				return dataRange
+			}),
+			Entry("required start is not set", func() *dexcom.DataRange {
+				dataRange := test.RandomDataRange()
+				dataRange.Start = nil
+				return dataRange
+			}),
+			Entry("required end display time is not set", func() *dexcom.DataRange {
+				dataRange := test.RandomDataRange()
+				dataRange.End.DisplayTime = nil
+				return dataRange
+			}),
+			Entry("required start system time is not set", func() *dexcom.DataRange {
+				dataRange := test.RandomDataRange()
+				dataRange.Start.SystemTime = nil
+				return dataRange
+			}),
+			Entry("required start display time is not set", func() *dexcom.DataRange {
+				dataRange := test.RandomDataRange()
+				dataRange.Start.DisplayTime = nil
+				return dataRange
+			}),
+		)
```

Comment:
Consider adding a test where it handles malformed time values. I suspect given the current code, they'll be silently dropped. Is that the behavior we want? I really don't know, I guess it could be, but I feel like that's an indicator that something's wrong and we might need to retry from the top.

Reply:
if the time is malformed the parsing would return an error and the the validation will fail as the required date isn't set. From memory finding the date issues wasn't too hard